discovery_engine/lib/src/ffi/types/document/document_vec.dart

```diff
@@ -15,13 +15,13 @@
 import 'dart:ffi' show Pointer;
 
 import 'package:xayn_discovery_engine/src/ffi/genesis.ffigen.dart'
-    show RustDocument, RustDocumentVec;
+    show RustDocument, RustVecDocument;
```

Comment:
Why this rename is necessary? It is really common to have `Vec`,`List`,`Buffer`,`Stream`, `Future` at the end.
Also with this rename all file names are out of sync.

Reply:
I would prefer the naming to change in that way too as it reads better for thinks like `Vec<u8>` (`VecU8` is readable `U8Vec` irritating, for me).

But I also think it's something we can do later.

The file names are out of sync because their are anyway sub-optimal. Like instead of `document_vec.rs` it probably should `document/vec.rs` or `vec/document.rs`.  Either way not something we should do now/this week.

Reply:
ok I see.For me `U8Vec` is better, like `Uint8List` but I prefer `OptionU8` over `U8Option`.